designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/ListCard.kt

```diff
@@ -0,0 +1,452 @@
+    modifier: Modifier = Modifier,
+) {
+    val expandableItemList = mutableListOf<AdditionalInfoItem>()
+    val constantItemList = mutableListOf<AdditionalInfoItem>()
+
+    additionalInfoList.forEach {
+            item ->
+        if (item.isConstantItem) {
+            constantItemList.add(item)
+        } else {
+            expandableItemList.add(item)
+        }
+    }
+    CompositionLocalProvider(LocalRippleTheme provides Ripple.CustomDHISRippleTheme) {
+        Row(
+            modifier = modifier
+                .background(color = TextColor.OnPrimary)
+                .clip(shape = RoundedCornerShape(Radius.S))
+                .clickable(onClick = onCardClick)
+                .padding(Spacing.Spacing8)
+                .hoverPointerIcon(true),
+        ) {
+            listAvatar?.let {
+                it.invoke()
+                Spacer(Modifier.size(Spacing.Spacing16))
+            }
+            Column(Modifier.fillMaxWidth().weight(1f)) {
+                Row(horizontalArrangement = Arrangement.SpaceBetween) {
+                    // Row with header and last updated
+                    ListCardTitle(text = title, modifier.weight(1f))
+                    if (lastUpdated != null) {
+                        ListCardLastUpdated(lastUpdated)
+                    }
+                }
+                AdditionalInfoColumn(
+                    expandableItems = expandableItemList,
+                    constantItems = constantItemList,
+                    modifier = Modifier.testTag("LIST_CARD_ADDITIONAL_INFO_COLUMN"),
+                    expandLabelText = expandLabelText,
+                    shrinkLabelText = shrinkLabelText,
+                    syncProgressItem = AdditionalInfoItem(
+                        icon = {
+                            ProgressIndicator(
+                                type = ProgressIndicatorType.CIRCULAR,
+                                hasError = false,
+                            )
+                        },
+                        value = "Syncing...",
+                        color = SurfaceColor.Primary,
+                        isConstantItem = false,
+                    ),
+                    showLoading = showLoading,
+
+                )
+                actionButton?.invoke()
+            }
+        }
+    }
+}
+
+/**
+ * DHIS2 ListCardDetail.
+ * Component intended for TEI Dashboard
+ * @param title is the card title
+ * @param additionalInfoList is a list of AdditionalInfoItem that
+ * manages all the key value types that will be shown
+ * if there are more than three items that are not constant
+ * a show more/less button will appear and the rest of items will be hidden
+ * @param expandLabelText the text to be shown for expand button
+ * @param shrinkLabelText the text to be shown for shrink button
+ * @param actionButton composable parameter for the sync button
+ * @param modifier allows a modifier to be passed externally
+ */
+@Composable
+fun ListCardDetail(
```

Comment:
The previous function is intended to render list items so it's called `ListCard`, as this function is intended to render TEI details, maybe we can remove the List and call it just `CardDetail`. What do you think?

Reply:
I agree naming could be improved